suzieq/poller/controller/credential_loader/static.py

```diff
@@ -34,53 +49,39 @@ def init(self, init_data: dict):
         Raises:
             InventorySourceError: Invalid env argument
         """
-        self._username = init_data.get('username')
-
-        if init_data.get('keyfile'):
-            self._keyfile = init_data['keyfile']
-
-        if init_data.get('password'):
-            password = init_data['password']
-            if password.startswith('env:'):
-                self._password = getenv(password.split('env:')[1], '')
-                if not self._password:
-                    raise InventorySourceError(
-                        f'No password in environment '
-                        f'variable "{password.split("env:")[1]}"')
-            elif password.startswith('plain:'):
-                self._password = password.split("plain:")[1]
-            elif password.startswith('ask'):
-                self._password = getpass.getpass(
-                    f'{self._name} Password to login to device: ')
-            else:
-                raise InventorySourceError(
-                    f'{self._name} unknown password method.'
-                    'Supported methods are ["ask", "plain:", "env:"]')
-
-        if init_data.get('ssh-passphrase'):
-            passphrase = init_data['ssh-passphrase']
-            if passphrase.startswith('env:'):
-                self._passphrase = getenv(
-                    passphrase.split('env:')[1], '')
-                if not self._passphrase:
-                    raise InventorySourceError(
-                        f'No passphrase in environment '
-                        f'variable "{passphrase.split("env:")[1]}"')
-            elif passphrase.startswith('plain:'):
-                self._passphrase = passphrase.split("plain:")[1]
-            elif passphrase.startswith('ask'):
-                self._passphrase = getpass.getpass(
-                    f'{self._name} Passphrase to decode private key file: '
+        if not self._validate:
+            # fix pydantic alias
+            init_data['ssh_passphrase'] = init_data.pop('ssh-passphrase', None)
+        super().init(init_data)
+
+        if self._data.password == 'ask':
+            try:
+                self._data.password = get_sensitive_data(
+                    self._data.password,
+                    f'{self.name} Password to login to device: ')
```

Comment:
I think here we should print also some device info to identify which one we are asking the password for (hostname, address, maybe even the namespace).

Reply:
It is not true. This is the static loader, so you are specifying a single password to access all the devices